src/api/src/create_book.py

```diff
@@ -143,6 +143,39 @@ async def fetch_cover(url: str, cookies: Optional[dict] = None) -> bytes:
     return body
 
 
+@backoff.on_exception(backoff.expo, ClientResponseError, max_time=15)
+async def fetch_story_ID(part_id: int, cookies: Optional[dict] = None) -> int:
+    """Return the Story ID given a Part ID"""
+    async with (
+        CachedSession(headers=headers, cache=cache)
+        if not cookies
+        else ClientSession(headers=headers, cookies=cookies)
+    ) as session:  # Don't cache requests with Cookies.
+        async with session.get(
+            f"https://www.wattpad.com/api/v3/story_parts/{part_id}?fields=url"
```

Comment:
@AaronBenDaniel Good stuff! How did you find this endpoint?

Reply:
I found the one for getting a Part URL from a Part ID in wattpad-py. In my searching I found the same method in wattpad-rs as well. The "endpoint" for getting a Story URL from a Part URL is just downloading the HTML content of the Part URL.

I asked myself, "how does a human find the Story URL from a Part URL" and realized that it's just in the top left-hand corner of the reading interface. That URL appears to be gotten directly from the Part URL and not the API, so the HTML content of the web reading interface contains the Story URL. This just fetches that content and extracts the Story URL.